pkg/k8shandler/elasticsearch.go

```diff
@@ -440,6 +440,123 @@ func DoSynchronizedFlush(clusterName, namespace string, client client.Client) (b
 	return (payload.StatusCode == 200), payload.Error
 }
 
+func MigrateKibanaIndex(clusterName, namespace string, client client.Client) (bool, error) {
+	// check that kibana needs to be migrated
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		URI:    ".kibana/_mapping",
+	}
+
+	curlESService(clusterName, namespace, payload, client)
+
+	// check for 404, means index does not exist -- no migration needed
+	if payload.StatusCode == 404 {
+		return true, nil
+	}
+
+	if indexBody, ok := payload.ResponseBody[".kibana"].(map[string]interface{}); ok {
+		if mappings, ok := indexBody["mappings"].(map[string]interface{}); ok {
+			if len(mappings) > 1 {
+				// if .kibana_1 doesn't exist, reindex
+				payload := &esCurlStruct{
+					Method: http.MethodGet,
```

Comment:
Actually, there is better [method to check if index exists](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/indices-exists.html). This should save some resources but we probably need to check first if our client is authorized to make this call.